streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsMetricsImpl.java

```diff
@@ -42,14 +42,15 @@
     final Metrics metrics;
     final String groupName;
     final Map<String, String> tags;
+    final Map<Sensor, Sensor> parentSensors;
 
     public StreamsMetricsImpl(Metrics metrics, String groupName,  Map<String, String> tags) {
         Objects.requireNonNull(metrics, "Metrics cannot be null");
 
         this.metrics = metrics;
         this.groupName = groupName;
         this.tags = tags;
-
+        this.parentSensors = new HashMap<>();
```

Comment:
@guozhangwang `addLatencyMetrics` is supposed to return 1 sensors, but it creates 2 internally. One of them, the parent, is never exposed to the user, so the user has no way of deleting it. With this solution, the user deletes the child sensor and we internally delete the parent sensor.